src/time_zone_format.cc

```diff
@@ -39,7 +39,7 @@
 #include <limits>
 #include <string>
 #include <vector>
-#if !HAS_STRPTIME
+#if !defined(HAS_STRPTIME)
```

Comment:
`#if !defined(HAS_STRPTIME)` is not the same as `#if !HAS_STRPTIME` when `HAS_STRPTIME=0`, which is intended to be a valid way to say "I don't have strptime()".

I see that we have `#if defined(HAS_STRPTIME) && HAS_STRPTIME` in another place (presumably to workaround the same warning), so I suggest we rework these new changes with the negation of that.

[All that said, I'm not a fan of `-Wundef`, but I guess that's a lost argument.]

Reply:
#331 addresses this comment.